src/shims/mod.rs

```diff
@@ -26,11 +27,6 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
             dest.map(|place| *place)
         );
 
-        // First, run the common hooks also supported by CTFE.
-        if this.hook_fn(instance, args, dest)? {
```

Comment:
We should probably rename this to `hook_panic_fn` on the rustc side... do you want to do that as part of our PR, or should I make a new one?

Reply:
I think it's fine leaving it as `hook_fn` - it just happens to only handle panic-related stuff at the moment.

Reply:
If it ever hooks anything else we'll likely want to also have that hooked in Miri. So I am proposing the rename precisely because I don't want other stuff to be handled there.

Reply:
I've renamed the mthod.